irohad/ametsuchi/impl/storage_impl.cpp

```diff
@@ -306,7 +306,7 @@ namespace iroha {
       auto init_db = [&](soci::session &session) {
         // rollback current prepared transaction
         // if there exists any since last session
-        if (prepared_blocks_enabled_) {
+        if (block_is_prepared_) {
```

Comment:
This condition will always be false, since `block_is_prepared_` is set to false in constructor. As it is said in the comment, there might be some prepared transaction from the previous run of the daemon, and it should be rolled back. Change should be reverted.
```suggestion
        if (prepared_blocks_enabled_) {
```